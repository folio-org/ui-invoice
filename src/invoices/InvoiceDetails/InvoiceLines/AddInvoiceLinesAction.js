import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { Pluggable } from '@folio/stripes/core';

const buttonLabel = <FormattedMessage id="ui-invoice.invoice.details.lines.add" />;

const AddInvoiceLinesAction = ({ addLines, isDisabled }) => {
  return (
    <Pluggable
      aria-haspopup="true"
      searchButtonStyle="default"
      searchLabel={buttonLabel}
      type="find-po-line"
      addLines={addLines}
      disabled={isDisabled}
    >
      <FormattedMessage id="ui-invoice.find-po-line-plugin-unavailable" />
    </Pluggable>
  );
};

AddInvoiceLinesAction.propTypes = {
  addLines: PropTypes.func.isRequired,
  isDisabled: PropTypes.bool,
};

AddInvoiceLinesAction.defaultProps = {
  isDisabled: false,
};

export default AddInvoiceLinesAction;
