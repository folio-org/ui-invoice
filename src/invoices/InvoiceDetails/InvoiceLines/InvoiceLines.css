.invoiceLinesActions {
  display: flex;

  & button {
    margin-left: 5px;
  }
}

.invoiceLinesTotal {
  font-weight: 600;
  padding: 0 15px;
}
