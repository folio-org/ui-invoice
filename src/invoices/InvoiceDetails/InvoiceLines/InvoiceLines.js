import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { get } from 'lodash';

import { MultiColumnList } from '@folio/stripes/components';
import { stripesConnect } from '@folio/stripes/core';

import {
  invoiceLinesResource,
} from '../../../common/resources';
import styles from './InvoiceLines.css';

const visibleColumns = ['description', 'invoiceLineNumber', 'quantity', 'subTotal'];
const columnMapping = {
  description: <FormattedMessage id="ui-invoice.invoice.details.lines.list.description" />,
  invoiceLineNumber: <FormattedMessage id="ui-invoice.invoice.details.lines.list.number" />,
  quantity: <FormattedMessage id="ui-invoice.invoice.details.lines.list.quantity" />,
  subTotal: <FormattedMessage id="ui-invoice.invoice.details.lines.list.total" />,
};
const columnWidths = {
  description: '40%',
  invoiceLineNumber: '20%',
  quantity: '20%',
  subTotal: '20%',
};

class InvoiceLines extends Component {
  static manifest = Object.freeze({
    invoiceLines: {
      ...invoiceLinesResource,
      GET: {
        params: {
          query: (queryParams, pathComponents, resourceValues) => {
            if (resourceValues.invoiceId && resourceValues.invoiceId.length) {
              return `(invoiceId==${resourceValues.invoiceId})`;
            }

            return null;
          },
        },
      },
    },
    invoiceId: {},
    query: {},
  });

  static propTypes = {
    invoiceId: PropTypes.string.isRequired,
    resources: PropTypes.object.isRequired,
    mutator: PropTypes.object.isRequired,
  };

  componentDidUpdate() {
    const { invoiceId, resources, mutator } = this.props;

    if (invoiceId !== resources.invoiceId) {
      mutator.invoiceId.replace(invoiceId);
    }
  }

  openLineDetails = (e, invoiceLine) => {
    const _path = `/invoice/view/${invoiceLine.invoiceId}/line/${invoiceLine.id}/view`;

    this.props.mutator.query.update({ _path });
  }

  render() {
    const { resources } = this.props;
    const invoiceLinesItems = get(resources, 'invoiceLines.records.0.invoiceLines', []);

    return (
      <Fragment>
        <div className={styles.invoiceLinesTotal}>
          <FormattedMessage
            id="ui-invoice.invoiceLine.total"
            values={{ total: invoiceLinesItems.length }}
          />
        </div>

        <MultiColumnList
          contentData={invoiceLinesItems}
          visibleColumns={visibleColumns}
          columnMapping={columnMapping}
          columnWidths={columnWidths}
          onRowClick={this.openLineDetails}
        />
      </Fragment>
    );
  }
}

export default stripesConnect(InvoiceLines);
