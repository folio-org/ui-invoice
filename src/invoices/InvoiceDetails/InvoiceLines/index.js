export { default as InvoiceLinesActions } from './InvoiceLinesActions';
export { default } from './InvoiceLinesContainer';
