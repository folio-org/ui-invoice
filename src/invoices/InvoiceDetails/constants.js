export const VENDOR_STATUS = {
  ACTIVE: 'Active',
  INACTIVE: 'Inactive',
  PENDING: 'Pending',
};

export const ACQ_ERROR_TYPE = {
  order: 'order',
};

export const INVOICE_OMITTED_FIELDS = [
  'approvalDate',
  'approvedBy',
  'documents',
  'fiscalYearId',
  'id',
  'links',
  'metadata',
  'paymentDate',
  'status',
  'voucherNumber',
];
export const PO_LINE_PAYMENT_STATUS = {
  NO_CHANGE: 'noChange',
  AWAITING_PAYMENT: 'awaitingPayment',
  PARTIALLY_PAID: 'partiallyPaid',
  FULLY_PAID: 'fullyPaid',
  CANCELLED: 'cancelled',
};

export const PO_LINE_PAYMENT_STATUSES = [
  PO_LINE_PAYMENT_STATUS.NO_CHANGE,
  PO_LINE_PAYMENT_STATUS.AWAITING_PAYMENT,
  PO_LINE_PAYMENT_STATUS.PARTIALLY_PAID,
  PO_LINE_PAYMENT_STATUS.FULLY_PAID,
  PO_LINE_PAYMENT_STATUS.CANCELLED,
];

export const PO_LINE_PAYMENT_STATUS_LABELS = {
  [PO_LINE_PAYMENT_STATUS.NO_CHANGE]: 'No Change',
  [PO_LINE_PAYMENT_STATUS.AWAITING_PAYMENT]: 'Awaiting Payment',
  [PO_LINE_PAYMENT_STATUS.PARTIALLY_PAID]: 'Partially Paid',
  [PO_LINE_PAYMENT_STATUS.FULLY_PAID]: 'Fully Paid',
  [PO_LINE_PAYMENT_STATUS.CANCELLED]: 'Cancelled',
};

export const ORDER_TYPE = {
  ONE_TIME: 'One-Time',
  ONGOING: 'Ongoing',
};
