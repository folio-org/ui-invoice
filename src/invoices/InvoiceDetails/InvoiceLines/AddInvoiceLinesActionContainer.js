import PropTypes from 'prop-types';
import {
  useCallback,
  useEffect,
  useState,
} from 'react';
import { FormattedMessage } from 'react-intl';

import { stripesConnect } from '@folio/stripes/core';
import { ConfirmationModal } from '@folio/stripes/components';
import {
  batchFetch,
  useModalToggle,
  useShowCallout,
  useToggle,
} from '@folio/stripes-acq-components';

import { CurrencyMismatchModal } from '../../../common/components';
import { ordersResource } from '../../../common/resources';
import AddInvoiceLinesAction from './AddInvoiceLinesAction';

export const AddInvoiceLinesActionContainerComponent = ({
  addLines,
  invoiceCurrency,
  invoiceVendorId,
  mutator,
  onClose,
}) => {
  const [isAddInvoiceLines, toggleAddInvoiceLines] = useToggle(true);
  const [isCurrencyConfirmation, toggleCurrencyConfirmation] = useModalToggle();
  const [isVendorConfirmation, toggleVendorConfirmation] = useModalToggle();
  const [poLines, setPoLines] = useState();
  const showCallout = useShowCallout();

  useEffect(() => {
    if (!(isAddInvoiceLines || poLines?.length)) {
      onClose();
    }
  }, [poLines, isAddInvoiceLines, onClose]);

  const getVendorsIds = useCallback(
    async (ordersIds) => {
      let vendorsIds;

      try {
        const ordersResponse = await batchFetch(mutator.orders, ordersIds);

        vendorsIds = ordersResponse.map(({ vendor }) => vendor);
      } catch (e) {
        setPoLines(undefined);
        showCallout({ messageId: 'ui-invoice.invoice.actions.addLine.vendorLoadError', type: 'error' });
      }

      return vendorsIds;
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [showCallout],
  );

  const validateCurrency = useCallback(
    (selectedPoLines) => {
      const currencies = selectedPoLines.map(line => line.cost.currency);

      return currencies.some(currency => currency !== invoiceCurrency)
        ? toggleCurrencyConfirmation()
        : addLines(selectedPoLines);
    },
    [invoiceCurrency, toggleCurrencyConfirmation, addLines],
  );

  const validateVendor = useCallback(
    async (selectedPoLines) => {
      const ordersIds = selectedPoLines.map(line => line.purchaseOrderId);
      const vendorsIds = await getVendorsIds(ordersIds);

      if (!vendorsIds) return null;

      return (
        vendorsIds.some(vendorId => vendorId !== invoiceVendorId)
          ? toggleVendorConfirmation()
          : validateCurrency(selectedPoLines)
      );
    },
    [invoiceVendorId, toggleVendorConfirmation, validateCurrency, getVendorsIds],
  );

  const validateSelectedLines = useCallback(
    (selectedPoLines) => {
      setPoLines(selectedPoLines);
      validateVendor(selectedPoLines);
    },
    [validateVendor],
  );

  return (
    <>
      {
        isAddInvoiceLines && (
          <AddInvoiceLinesAction
            addLines={addLines}
            onClose={toggleAddInvoiceLines}
            validateSelectedRecords={validateSelectedLines}
          />
        )
      }

      <CurrencyMismatchModal
        onCancel={() => {
          toggleCurrencyConfirmation();
          setPoLines(undefined);
        }}
        onConfirm={() => {
          toggleCurrencyConfirmation();
          addLines(poLines);
          setPoLines(undefined);
        }}
        open={isCurrencyConfirmation}
      />

      {isVendorConfirmation && (
        <ConfirmationModal
          id="invoice-line-vendor-confirmation"
          confirmLabel={<FormattedMessage id="ui-invoice.invoice.actions.addLine.confirmLabel" />}
          heading={<FormattedMessage id="ui-invoice.invoice.actions.addLine.heading" />}
          message={<FormattedMessage id="ui-invoice.invoice.actions.addLine.vendorMessage" />}
          onCancel={() => {
            toggleVendorConfirmation();
            setPoLines(undefined);
          }}
          onConfirm={() => {
            toggleVendorConfirmation();
            validateCurrency(poLines);
          }}
          open
        />
      )}
    </>
  );
};

AddInvoiceLinesActionContainerComponent.manifest = Object.freeze({
  orders: ordersResource,
});

AddInvoiceLinesActionContainerComponent.propTypes = {
  addLines: PropTypes.func.isRequired,
  invoiceCurrency: PropTypes.string.isRequired,
  invoiceVendorId: PropTypes.string.isRequired,
  mutator: PropTypes.object.isRequired,
  onClose: PropTypes.func,
};

export default stripesConnect(AddInvoiceLinesActionContainerComponent);
