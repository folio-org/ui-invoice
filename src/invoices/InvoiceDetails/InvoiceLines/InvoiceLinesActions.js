import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import {
  Button,
} from '@folio/stripes/components';
import { IfPermission } from '@folio/stripes/core';

import AddInvoiceLinesAction from './AddInvoiceLinesAction';
import styles from './InvoiceLines.css';

const InvoiceLinesActions = ({ addLines, createLine, isDisabled }) => (
  <IfPermission perm="invoice.invoice-lines.item.post">
    <div className={styles.invoiceLinesActions}>
      <AddInvoiceLinesAction
        addLines={addLines}
        isDisabled={isDisabled}
      />

      <Button
        data-test-button-create-line
        onClick={createLine}
        disabled={isDisabled}
      >
        <FormattedMessage id="ui-invoice.button.createLine" />
      </Button>
    </div>
  </IfPermission>
);

InvoiceLinesActions.propTypes = {
  addLines: PropTypes.func.isRequired,
  createLine: PropTypes.func.isRequired,
  isDisabled: PropTypes.bool,
};

InvoiceLinesActions.defaultProps = {
  isDisabled: false,
};

export default InvoiceLinesActions;
