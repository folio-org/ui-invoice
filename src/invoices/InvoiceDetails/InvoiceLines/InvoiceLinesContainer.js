import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import ReactRouterPropTypes from 'react-router-prop-types';

import {
  useInvoiceOrders,
} from './hooks';
import InvoiceLines from './InvoiceLines';

export const InvoiceLinesContainerComponent = ({
  invoiceLines,
  invoice,
  vendor,
  history,
  location,
  orderlinesMap,
  refreshData,
  visibleColumns,
}) => {
  const { orders } = useInvoiceOrders(invoice);

  const openLineDetails = useCallback(
    (e, invoiceLine) => {
      const pathname = `/invoice/view/${invoiceLine.invoiceId}/line/${invoiceLine.id}/view`;

      history.push({
        pathname,
        search: location.search,
      });
    },
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [location],
  );

  return (
    <InvoiceLines
      invoice={invoice}
      vendor={vendor}
      orders={orders}
      invoiceLinesItems={invoiceLines}
      openLineDetails={openLineDetails}
      orderlinesMap={orderlinesMap}
      refreshData={refreshData}
      visibleColumns={visibleColumns}
    />
  );
};

InvoiceLinesContainerComponent.propTypes = {
  invoice: PropTypes.object.isRequired,
  vendor: PropTypes.object.isRequired,
  history: ReactRouterPropTypes.history.isRequired,
  location: ReactRouterPropTypes.location.isRequired,
  invoiceLines: PropTypes.arrayOf(PropTypes.object),
  orderlinesMap: PropTypes.object,
  refreshData: PropTypes.func.isRequired,
  visibleColumns: PropTypes.arrayOf(PropTypes.string),
};

InvoiceLinesContainerComponent.defaultProps = {
  invoiceLines: [],
};

export default withRouter(InvoiceLinesContainerComponent);
