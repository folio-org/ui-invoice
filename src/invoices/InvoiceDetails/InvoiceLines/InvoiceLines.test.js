import { MemoryRouter } from 'react-router-dom';

import { render } from '@folio/jest-config-stripes/testing-library/react';

import { invoice, invoiceLine } from '../../../../test/jest/fixtures';

import { INVOICE_LINES_COLUMN_MAPPING } from '../../constants';
import InvoiceLines from './InvoiceLines';

jest.mock('./InvoiceLineOrderLineNumber', () => ({
  InvoiceLineOrderLineNumber: jest.fn(() => 'InvoiceLineOrderLineNumber'),
}));
jest.mock('./InvoiceLineOrderLineLink', () => ({
  InvoiceLineOrderLineLink: jest.fn(() => 'InvoiceLineOrderLineLink'),
}));

const defaultProps = {
  invoice,
  invoiceLinesItems: [invoiceLine],
  orderlinesMap: {},
  vendor: {},
  visibleColumns: Object.keys(INVOICE_LINES_COLUMN_MAPPING),
  openLineDetails: jest.fn(),
  refreshData: jest.fn(),
};
const renderInvoiceLines = (props = defaultProps) => render(
  <InvoiceLines {...props} />,
  { wrapper: MemoryRouter },
);

describe('InvoiceLines', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure', async () => {
    const { asFragment } = renderInvoiceLines();

    expect(asFragment()).toMatchSnapshot();
  });
});
