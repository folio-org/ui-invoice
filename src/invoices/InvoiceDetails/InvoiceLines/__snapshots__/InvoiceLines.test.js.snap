// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceLines should render correct structure 1`] = `
<DocumentFragment>
  <div
    class="invoiceLinesTotal"
  >
    ui-invoice.invoiceLine.total
  </div>
  <div
    aria-live="assertive"
    aria-relevant="additions"
    class="sr-only"
    tabindex="-1"
  />
  <div
    aria-rowcount="2"
    class="mclContainer"
    data-total-count="0"
    id="invoice-lines-list"
    role="grid"
    style="position: relative; overflow: hidden; height: auto; width: 100%;"
    tabindex="0"
  >
    <div
      class="mclHeaderContainer"
    >
      <div
        aria-rowindex="1"
        class="mclHeaderRow"
        role="row"
        style="display: flex;"
      >
        <div
          aria-sort="ascending"
          class="mclHeader mclClickable mclSorted mclAscending"
          id="list-column-linenumber"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          <div
            class="mclHeaderOuter"
            data-test-clickable-header="true"
            id="clickable-list-column-linenumber"
            role="button"
            tabindex="0"
          >
            <div
              class="mclHeaderInner"
            >
              <div
                style="max-width: calc(100%-15px);"
              >
                ui-invoice.invoice.details.lines.list.lineNumber
              </div>
            </div>
          </div>
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-polnumber"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.lines.list.polNumber
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-description"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.lines.list.description
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-fundcode"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.lines.list.fundCode
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-postatus"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.lines.list.poStatus
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-receiptstatus"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.lines.list.receiptStatus
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-paymentstatus"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.lines.list.paymentStatus
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-vendorrefno"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.lines.list.vendorRefNumber
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-quantity"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.lines.list.quantity
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-subtotal"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.lines.list.subTotal
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-adjustmentstotal"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.lines.list.adjustments
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-total"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.lines.list.total
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-vendorcode"
          role="columnheader"
          style="flex: 0 0 auto;"
        >
          ui-invoice.invoice.details.vendor.code
        </div>
        <div
          aria-sort="none"
          class="mclHeader"
          id="list-column-arrow"
          role="columnheader"
          style="flex: 0 0 auto;"
        />
      </div>
    </div>
    <div
      class="mclScrollable"
      style="position: static; width: 100%;"
      tabindex="0"
    >
      <div
        class="mclRowContainer"
        role="rowgroup"
        style="position: static;"
      >
        <div
          aria-rowindex="2"
          class="mclRowFormatterContainer"
          data-row-index="row-0"
          role="row"
          style="position: static;"
        >
          <div
            aria-label=""
            class="mclRow mclIsOdd mclIsInteractive mclClickableRow alignLastColToEnd"
            data-row-inner="0"
            style="min-width: 0px;"
            tabindex="0"
          >
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              <span
                id="729c379e-ed65-4e8b-ac28-3d75386a5ffc"
              />
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              InvoiceLineOrderLineNumber
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              EDI import invoice line
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              <span>
                -
              </span>
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              <span>
                -
              </span>
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              <span>
                -
              </span>
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              <span>
                -
              </span>
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              <span>
                -
              </span>
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              20
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              $1,523.00
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              <span>
                -
              </span>
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              <span>
                -
              </span>
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              <span>
                -
              </span>
            </div>
            <div
              class="mclCell  mclCellStyle"
              role="gridcell"
            >
              <span
                data-testid="arrow-0"
              >
                Icon
              </span>
            </div>
          </div>
        </div>
        <div
          class="mclEndOfListContainer"
          data-end-of-list="1"
        >
          <div
            class="mclCenteredContainer"
            style="width: 100%; visibility: hidden; height: 0px; padding: 0px;"
          >
            <div
              class="textCentered"
              data-test-layout="true"
            >
              <div
                class="mclEndOfList"
                data-test-layout="true"
              >
                <span>
                  Icon
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="textCentered noPaddingBottom"
    data-test-layout="true"
  >
    <div
      class="mclEndOfList"
      data-test-layout="true"
    >
      <span>
        Icon
      </span>
    </div>
  </div>
  InvoiceLineOrderLineLink
</DocumentFragment>
`;
